pkg/dataobj/metastore/metastore.go

```diff
@@ -1,341 +1,30 @@
 package metastore
 
 import (
-	"bytes"
 	"context"
-	"fmt"
-	"io"
-	"iter"
-	"sort"
-	"strconv"
-	"sync"
 	"time"
 
-	"github.com/go-kit/log"
-	"github.com/go-kit/log/level"
-	"github.com/grafana/dskit/backoff"
-	"github.com/pkg/errors"
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/prometheus/model/labels"
-	"github.com/thanos-io/objstore"
-	"golang.org/x/sync/errgroup"
-
-	"github.com/grafana/loki/v3/pkg/dataobj"
-	"github.com/grafana/loki/v3/pkg/logproto"
-)
-
-const (
-	metastoreWindowSize = 12 * time.Hour
 )
 
-// Define our own builder config because metastore objects are significantly smaller.
-var metastoreBuilderCfg = dataobj.BuilderConfig{
-	TargetObjectSize:  32 * 1024 * 1024,
-	TargetPageSize:    4 * 1024 * 1024,
-	BufferSize:        32 * 1024 * 1024, // 8x page size
-	TargetSectionSize: 4 * 1024 * 1024,  // object size / 8
-}
-
-type Manager struct {
-	metastoreBuilder *dataobj.Builder
-	tenantID         string
-	metrics          *metastoreMetrics
-	bucket           objstore.Bucket
-	logger           log.Logger
-	backoff          *backoff.Backoff
-	buf              *bytes.Buffer
-
-	builderOnce sync.Once
-}
-
-func NewManager(bucket objstore.Bucket, tenantID string, logger log.Logger) *Manager {
-	metrics := newMetastoreMetrics()
-
-	return &Manager{
-		bucket:   bucket,
-		metrics:  metrics,
-		logger:   logger,
-		tenantID: tenantID,
-		backoff: backoff.New(context.TODO(), backoff.Config{
-			MinBackoff: 50 * time.Millisecond,
-			MaxBackoff: 10 * time.Second,
-		}),
-		builderOnce: sync.Once{},
-	}
-}
-
-func (m *Manager) RegisterMetrics(reg prometheus.Registerer) error {
-	return m.metrics.register(reg)
+type StreamMeta struct {
+	Labels *labels.Labels
 }
 
-func (m *Manager) UnregisterMetrics(reg prometheus.Registerer) {
-	m.metrics.unregister(reg)
+type DataObjectMeta struct {
```

Comment:
I added these wrapper objects for the interface in case we need to change the contents but I don't know if its a good idea since they currently don't contain more than 1 field. Should I keep it this way?

Reply:
I think we can avoid the wrapper for now, since they are one field only, when we have the need to multiple fields, then we add it

Reply:
No I don't at this point. We could possibly include things like the stream hash pre-calculated but I don't think thats particularly useful. I'll remove the wrappers and just return the labels & data object paths directly.